pynq_composable/composable.py

```diff
@@ -81,6 +81,32 @@ def _get_ip_name_by_vlnv(description: dict, vlnv: str) -> str:
             return k
     return None
 
+def _streamline_pipeline(pipe: list) -> list:
+    """
+    Flattens List
+    """
+    linear_pipe = [[], ]
+    for idx in range(len(pipe)):
+        element = pipe[idx]
+        if not isinstance(element, list):
+            linear_pipe[0].append(element)
+        else:
+            if len(element) < 2:
+                raise ValueError("branch needs at least two elements")
+            for idx2 in range(len(element)):
+                if idx2 > 0:
+                    linear_pipe.append([pipe[idx - 1]])
+                res = _streamline_pipeline(element[idx2])
+                for iii in range(len(res)):
+                    if iii > 0:
+                        if not res[iii] == [1]:
+                            linear_pipe.append(res[iii])
+                    else:
+                        if not res[iii] == [1]:
+                            linear_pipe[idx2].extend(res[iii])
```

Comment:
This seems too complex to remove '1' from the list.
You should try to remove the '1' when dealing with linear pipelines line 90